crates/csslsrs/src/features/hover.rs

```diff
@@ -111,104 +112,171 @@ fn get_css_hover_content(
+        writeln!(content,
+            "[Selector Specificity](https://developer.mozilla.org/docs/Web/CSS/Specificity): ({}, {}, {})\n",
             ids, classes, elements
-        ));
+        ).unwrap();
     }
+    content
+}
 
-    // Add restriction if available
-    if let Some(restriction) = restrictions {
-        content.push_str("**Restriction**:\n");
-        for restriction in restriction {
-            content.push_str(&format!("- {}\n", restriction.trim()));
+/// Escapes Markdown characters in the selector name.
+fn escape_markdown(text: &str) -> String {
+    text.replace('*', "\\*")
+        .replace('_', "\\_")
+        .replace('`', "\\`")
+        .replace('[', "\\[")
+        .replace(']', "\\]")
+        .replace('(', "\\(")
+        .replace(')', "\\)")
+}
+
+/// Parses the status of a CSS Custom Data to a human-readable string, and writes it to the hover content.
+fn write_status(content: &mut String, status: &Option<Status>) {
+    match status {
+        Some(Status::Experimental) => {
+            writeln!(content, "🧪 *Experimental, use with caution.*\n").unwrap()
+        }
+        Some(Status::Obsolete) => {
+            writeln!(content, "🚧 *Obsolete, consider using alternatives.*\n").unwrap()
         }
+        Some(Status::Nonstandard) => {
+            writeln!(content, "🚨 *Non-standard, avoid using it.*\n").unwrap()
+        }
+        _ => {}
+    }
+}
+
+/// Writes the description of a CSS Custom Data to the hover content.
+fn write_description(content: &mut String, description: &Option<MarkupDescriptionOrString>) {
+    if let Some(desc) = description {
+        let desc_str = match desc {
+            MarkupDescriptionOrString::MarkupDescription(md) => &md.value,
+            MarkupDescriptionOrString::String(s) => s,
+        };
+        writeln!(content, "{}\n", desc_str).unwrap();
     }
+}
 
-    // Add browsers if available
-    if let Some(browsers) = browsers {
-        content.push_str("**Supported Browsers**:\n");
-        for browser in browsers {
-            content.push_str(&format!("- {}\n", browser.trim()));
-        }
-        content.push('\n');
+/// Writes the syntax of a CSS Custom Data to the hover content.
+fn write_syntax(content: &mut String, syntax: Option<&str>) {
+    if let Some(syntax) = syntax {
+        writeln!(content, "Syntax: `{}`\n", syntax).unwrap();
     }
+}
 
-    // Add reference if available
+/// Writes the references of a CSS Custom Data to the hover content.
+fn write_references(
+    content: &mut String,
+    references: Option<&[Reference]>,
+    name_for_caniuse: &str,
+) {
     if let Some(references) = references {
-        content.push_str("**Reference**:\n");
+        let reference_str = references
+            .iter()
+            .map(|reference| format!("[{}]({})", reference.name, reference.url))
+            .collect::<Vec<_>>()
+            .join(", ");
+        // Add Can I Use as a bonus 🎁
```

Comment:
bit of a philosophical bit, but it's important imo to consider that features we have over other solutions are not, like, bonus or just cool things to do. They're actual part of our feature set. Users basically have no expectations here, since editor tooling often goes unnoticed